Adding Gamepad device discovery support to qtbase.

Temporary solution, as this will be part of future Qt releases.

diff -Naur qtbase-opensource-src-5.4.2_original/src/platformsupport/devicediscovery/qdevicediscovery_p.h qtbase-opensource-src-5.4.2/src/platformsupport/devicediscovery/qdevicediscovery_p.h
--- qtbase-opensource-src-5.4.2_original/src/platformsupport/devicediscovery/qdevicediscovery_p.h	2015-05-29 21:30:28.000000000 +0100
+++ qtbase-opensource-src-5.4.2/src/platformsupport/devicediscovery/qdevicediscovery_p.h	2015-06-24 10:04:48.202699105 +0100
@@ -74,7 +74,8 @@
         Device_DRM = 0x10,
         Device_DRM_PrimaryGPU = 0x20,
         Device_Tablet = 0x40,
-        Device_InputMask = Device_Mouse | Device_Touchpad | Device_Touchscreen | Device_Keyboard | Device_Tablet,
+        Device_Joystick = 0x80,
+        Device_InputMask = Device_Mouse | Device_Touchpad | Device_Touchscreen | Device_Keyboard | Device_Tablet | Device_Joystick,
         Device_VideoMask = Device_DRM
     };
     Q_DECLARE_FLAGS(QDeviceTypes, QDeviceType)
diff -Naur qtbase-opensource-src-5.4.2_original/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp qtbase-opensource-src-5.4.2/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp
--- qtbase-opensource-src-5.4.2_original/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp	2015-05-29 21:30:28.000000000 +0100
+++ qtbase-opensource-src-5.4.2/src/platformsupport/devicediscovery/qdevicediscovery_udev.cpp	2015-06-24 10:04:02.090360473 +0100
@@ -128,6 +128,8 @@
     }
     if (m_types & Device_Tablet)
         udev_enumerate_add_match_property(ue, "ID_INPUT_TABLET", "1");
+    if (m_types & Device_Joystick)
+        udev_enumerate_add_match_property(ue, "ID_INPUT_JOYSTICK", "1");
 
     if (udev_enumerate_scan_devices(ue) != 0) {
 #ifdef QT_QPA_DEVICE_DISCOVERY_DEBUG
@@ -251,6 +253,9 @@
     if ((m_types & Device_Tablet) && (qstrcmp(udev_device_get_property_value(dev, "ID_INPUT_TABLET"), "1") == 0))
         return true;
 
+    if ((m_types & Device_Joystick) && (qstrcmp(udev_device_get_property_value(dev, "ID_INPUT_JOYSTICK"), "1") == 0))
+        return true;
+
     if ((m_types & Device_DRM) && (qstrcmp(udev_device_get_subsystem(dev), "drm") == 0))
         return true;
 
